### PR TITLE
V2 sys metrics

### PR DESCRIPTION
**Pull request by pmcollins**
Add assertion for system metrics instrumentor. May address https://github.com/signalfx/splunk-otel-python/issues/557.

### CODE REVIEW

File: tests/ott_spec.py [hunk 1]
```diff
@@ -19,19 +22,26 @@ def on_start(self):
...
 
-        assert get_attribute(attributes, "process.pid")
+def assert_traces_to_spec(telemetry):
+    attrs = extract_leaves(telemetry, "trace_requests", "pbreq", "resource_spans", "resource", "attributes")
+    assert get_attribute(attrs, "telemetry.sdk.name")
+    assert get_attribute(attrs, "telemetry.sdk.version")
+    assert get_attribute(attrs, "telemetry.sdk.language")
+    assert get_attribute_str(attrs, "telemetry.distro.version")
+    assert get_attribute_str(attrs, "telemetry.distro.name") == "splunk-opentelemetry"
+    assert get_attribute(attrs, "process.pid")
```
Comment (pellared) [inline tests/ott_spec.py:39]: IMO this is not necessary. See https://github.com/signalfx/splunk-otel-python/issues/557#issuecomment-2513780042

File: tests/ott_spec.py [hunk 2]
```diff
@@ -10,7 +10,10 @@ def requirements(self):
         return project_path(), "oteltest"
 
     def environment_variables(self):
-        return {"OTEL_SERVICE_NAME": "my-svc"}
+        return {
+            "OTEL_SERVICE_NAME": "my-svc",
+            "OTEL_METRICS_ENABLED": "true",
```
Comment (pellared) [inline tests/ott_spec.py:15]: We must not introduce an env var with `OTEL_` prefix. These are reserved for OpenTelemetry (upstream)
Do we even need such configuration?
This is what we have in Go https://docs.splunk.com/observability/en/gdi/get-data-in/application/go/configuration/advanced-go-otel-configuration.html#exporters-configuration:

> `OTEL_METRICS_EXPORTER` - The metrics exporter to use. The default value is `otlp`. Accepted values are `otlp` and `none`. Setting `none` deactivates metric exports.

**Review by pellared — COMMENTED** [review]
I added "off-topic" kind of comments that are loosely related to the changes in this PR.

File: tests/ott_spec.py (hunk 1 above)
Comment (pmcollins) [reply tests/ott_spec.py:39]: Thx. I'll move this to another test that's not about the spec.

File: tests/ott_spec.py (hunk 2 above)
Comment (pmcollins) [reply tests/ott_spec.py:15]: Thanks -- these env vars are carried over from 1.x. I'm looking into removing at least some of them.

File: tests/ott_spec.py (hunk 1 above)
Comment (pellared) [reply tests/ott_spec.py]: > `process.pid` is in [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.18.0/specification/resource/semantic_conventions/process.md#process), so maybe we should keep it?

You are right! This one is important. There is something which takes advantage of it (this was the reason it was added it to the GDI spec).

File: tests/ott_sysmetrics.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class SysMetricsOtelTest:
+    def requirements(self):
+        return (project_path(),)
+
+    def environment_variables(self):
+        return {
+            "OTEL_SERVICE_NAME": "my-svc",
+            "OTEL_METRICS_ENABLED": "true",
```
Comment (pellared) [inline tests/ott_sysmetrics.py:14]: I think it is `true` by default so this line could be removed. Especially in the light of https://github.com/signalfx/splunk-otel-python/pull/565#discussion_r1867782025

```suggestion
```

**Comment by pmcollins** [comment]
I'm leaning towards doing away with `OTEL_METRICS_ENABLED` entirely. Users could still prevent the system metrics instrumentor from running via `OTEL_PYTHON_DISABLED_INSTRUMENTATIONS`. Any objections to this? cc: @johnbley

**Review by johnbley — APPROVED** [review]

**Pull request merged**